OMG.Classroom/DataAccessLayer/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using DataAccessLayer.Entities;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Repositories
+{
+    public abstract class BaseRepository<E> where E : BaseEntity
+    {
+        public readonly ClassroomDbContext _context;
+        public readonly DbSet<E> _dbSet;
+
+        protected BaseRepository(ClassroomDbContext context, DbSet<E> dbSet)
+        {
+            _context = context;
+            _dbSet = dbSet;
+        }
+
+        public async Task Add(E entity)
```

Comment:
Usually, when a method is asynchronous, the naming convention is MethodNameAsync. Change it everywhere